modules/sdk-coin-atom/src/lib/transaction.ts

```diff
@@ -0,0 +1,193 @@
+import {
+  BaseKey,
+  BaseTransaction,
+  InvalidTransactionError,
+  ParseTransactionError,
+  PublicKey as BasePublicKey,
+  Signature,
+  TransactionRecipient,
+  TransactionType,
+} from '@bitgo/sdk-core';
+import { AtomTransaction, GasFeeLimitData, MessageData, TransactionExplanation, TxData } from './iface';
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { Buffer } from 'buffer';
+import { AtomTransactionType, sendMsgType, thetaTestnetChainId } from './constants';
+import { SignerData, Coin, defaultRegistryTypes } from '@cosmjs/stargate';
+import { fromBase64 } from '@cosmjs/encoding';
+import { decodeTxRaw, EncodeObject, Registry } from '@cosmjs/proto-signing';
+
+export class Transaction extends BaseTransaction {
+  private _atomTransaction: AtomTransaction;
+  private _signature: Signature;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+  }
+
+  get atomTransaction(): AtomTransaction {
```

Comment:
Would be great if we had something like [Lombok](https://projectlombok.org/features/GetterSetter) in typescript to reduce boilerplate and look easy on the eye, but I'm not sure if there's a decent library that does that

Reply:
This will be abstracted more anyways via the transaction builder and the factory.